Source/WebKit/Platform/ios/VideoPresentationInterfaceLMK.mm

```diff
@@ -136,18 +140,34 @@ - (void)layoutSublayers
         if (auto* playbackSessionModel = this->playbackSessionModel()) {
             playbackSessionModel->setSpatialTrackingLabel(m_spatialTrackingLabel);
             playbackSessionModel->setSoundStageSize(WebCore::AudioSessionSoundStageSize::Large);
+
+            playableViewController().prefersAutoDimming = playbackSessionModel->prefersAutoDimming();
+
+            m_savedAutoDimmingPreference = playableViewController().prefersAutoDimming;
         }
         completionHandler(success, error);
     }).get()];
 }
 
+void VideoPresentationInterfaceLMK::prefersAutoDimmingChanged(bool prefersAutoDimming)
+{
+    if (prefersAutoDimming == m_savedAutoDimmingPreference)
+        return;
+
+    m_savedAutoDimmingPreference = prefersAutoDimming;
+    [m_playerViewController setPrefersAutoDimming:prefersAutoDimming];
+}
+
 void VideoPresentationInterfaceLMK::dismissFullscreen(bool animated, Function<void(BOOL, NSError *)>&& completionHandler)
 {
     playbackSessionInterface().stopObservingNowPlayingMetadata();
     [linearMediaPlayer() exitFullscreenWithCompletionHandler:makeBlockPtr([this, protectedThis = Ref { *this }, completionHandler = WTFMove(completionHandler)] (BOOL success, NSError *error) {
         if (auto* playbackSessionModel = this->playbackSessionModel()) {
             playbackSessionModel->setSpatialTrackingLabel(nullString());
             playbackSessionModel->setSoundStageSize(WebCore::AudioSessionSoundStageSize::Automatic);
+
+            m_savedAutoDimmingPreference = playableViewController().prefersAutoDimming;
```

Comment:
Is this even needed?

can prefersAutoDimming change ? I don't think we should store it here, considering we set it ourselves earlier, as such it serves no purpose.